charts/kubernikus-system/charts/prometheus/templates/_prometheus.yaml.tpl

```diff
@@ -241,7 +241,44 @@ scrape_configs:
       target_label: __metrics_path__
       replacement: /api/v1/nodes/${1}:4194/proxy/metrics
 
-# Static Targets
+- job_name: 'blackbox-ingress'
+  metrics_path: /probe
+  params:
+    # Look for a HTTP 200 response per default.
+    # Can be overwritten by annotating the ingress resource with the expected return codes, e.g. `prometheus.io/probe_code: "4xx"`
+    module: [http_2xx]
+  kubernetes_sd_configs:
+  - role: ingress
+  relabel_configs:
+  # don't scrape yourself
+  - source_labels: [__meta_kubernetes_ingress_path]
+    regex: /prometheus
+    action: drop
+  - source_labels: [__meta_kubernetes_ingress_annotation_prometheus_io_probe]
+    action: keep
+    regex: true
+  # consider prometheus.io/probe_code annotation. mind below regex.
+  - source_labels: [__meta_kubernetes_ingress_annotation_prometheus_io_probe_code]
+    regex: ^(\d).+
```

Comment:
I'm a bit confused about `http_2xx` and this code. Isn't this the expected code, that the proper should return? Or is this just for selecting the http-module?

Reply:
Both. You specify the module, like `http_2xx` and that defines which response codes are accepted. The blackbox exporter receives that as a query parameter.